python/cugraph/cugraph/tests/mg/test_mg_egonet.py

```diff
@@ -0,0 +1,145 @@
+from cudf.testing.testing import assert_frame_equal, assert_series_equal
+
+
+# =============================================================================
+# Pytest Setup / Teardown - called for each test function
+# =============================================================================
+
+
+def setup_function():
+    gc.collect()
+
+
+IS_DIRECTED = [True, False]
+SEEDS = [0, 5, 13, [0, 2]]
+RADIUS = [1, 2, 3]
+
+
+# =============================================================================
+# Pytest fixtures
+# =============================================================================
+
+datasets = utils.DATASETS_UNDIRECTED + [
+    utils.RAPIDS_DATASET_ROOT_DIR_PATH / "email-Eu-core.csv"
+]
+
+fixture_params = utils.genFixtureParamsProduct(
+    (datasets, "graph_file"),
+    (IS_DIRECTED, "directed"),
+    (SEEDS, "seeds"),
+    (RADIUS, "radius"),
+)
+
+
+@pytest.fixture(scope="module", params=fixture_params)
+def input_combo(request):
+    """
+    Simply return the current combination of params as a dictionary for use in
+    tests or other parameterized fixtures.
+    """
+    parameters = dict(zip(("graph_file", "directed", "seeds", "radius"), request.param))
+
+    return parameters
+
+
+@pytest.fixture(scope="module")
+def input_expected_output(input_combo):
+    """
+    This fixture returns the inputs and expected results from the egonet algo.
+    (based on cuGraph batched_ego_graphs) which can be used for validation.
+    """
+
+    input_data_path = input_combo["graph_file"]
+    directed = input_combo["directed"]
+    seeds = input_combo["seeds"]
+    radius = input_combo["radius"]
+    G = utils.generate_cugraph_graph_from_file(input_data_path, directed=directed)
+
+    sg_cugraph_ego_graphs = cugraph.batched_ego_graphs(G, seeds=seeds, radius=radius)
+
+    # Save the results back to the input_combo dictionary to prevent redundant
+    # cuGraph runs. Other tests using the input_combo fixture will look for
+    # them, and if not present they will have to re-run the same cuGraph call.
+
+    input_combo["sg_cugraph_results"] = sg_cugraph_ego_graphs
+    chunksize = dcg.get_chunksize(input_data_path)
+    ddf = dask_cudf.read_csv(
+        input_data_path,
+        chunksize=chunksize,
+        delimiter=" ",
+        names=["src", "dst", "value"],
+        dtype=["int32", "int32", "float32"],
+    )
+
+    dg = cugraph.Graph(directed=directed)
+    dg.from_dask_cudf_edgelist(
+        ddf,
+        source="src",
+        destination="dst",
+        edge_attr="value",
+        renumber=True,
+        legacy_renum_only=True,
+        store_transposed=True,
+    )
+
+    input_combo["MGGraph"] = dg
+
+    return input_combo
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+
+
+# @pytest.mark.skipif(
+#    is_single_gpu(), reason="skipping MG testing on Single GPU system"
+# )
```

Comment:
Don't we want this uncommented for local SG runs?

Reply:
Local SG runs will skip them if we exclude the `mg` (which is usually what happens) but I can add it